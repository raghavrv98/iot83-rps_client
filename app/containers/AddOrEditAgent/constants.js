/*
 *
 * AddOrEditAgent constants
 *
 */

export const CREATE_AGENT_REQUEST = "app/AddOrEditAgent/CREATE_AGENT_REQUEST";
export const CREATE_AGENT_SUCCESS = "app/AddOrEditAgent/CREATE_AGENT_SUCCESS";
export const CREATE_AGENT_FAILURE = "app/AddOrEditAgent/CREATE_AGENT_FAILURE";

export const GET_AGENT_DETAILS = "app/AddOrEditAgent/GET_AGENT_DETAILS";
export const GET_AGENT_DETAILS_SUCCESS = "app/AddOrEditAgent/GET_AGENT_DETAILS_SUCCESS";
export const GET_AGENT_DETAILS_FAILURE = "app/AddOrEditAgent/GET_AGENT_DETAILS_FAILURE";
