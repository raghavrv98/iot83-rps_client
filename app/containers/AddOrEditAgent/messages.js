/*
 * AddOrEditAgent Messages
 *
 * This contains all the text for the AddOrEditAgent container.
 */

import { defineMessages } from "react-intl";

export const scope = "app.containers.AddOrEditAgent";

export default defineMessages({
  headingAdd: {
    id :`${scope}.headingAdd`,
    defaultMessage : "Create Agent"
  },
  deviceId: {
    id :`${scope}.deviceId`,
    defaultMessage : "Device Id"
  },
  headingEdit: {
    id :`${scope}.headingEdit`,
    defaultMessage : "Edit Agent"
  },
  option1: {
    id :`${scope}.option1`,
    defaultMessage : "AP Sensing DTS Poller"
  },
  option2: {
    id :`${scope}.option2`,
    defaultMessage : "DCS Agent"
  },
  modbusPort: {
    id :`${scope}.addModbusPort`,
    defaultMessage : "Modbus Port"
  },
  addModbusMap: {
    id :`${scope}.addModbusMap`,
    defaultMessage : "Modbus Map"
  },
  agentUpdateMessage: {
    id :`${scope}.agentUpdateMessage`,
    defaultMessage : "Agent updated successfully."
  },
  agentCreateMessage: {
    id :`${scope}.agentCreateMessage`,
    defaultMessage : "Agent added successfully."
  },
  ipAddress: {
    id :`${scope}.ipAddress`,
    defaultMessage : "IP Address"
  },
  instrumentSetting: {
    id :`${scope}.instrumentSetting`,
    defaultMessage : "Instrument Setting"
  },
  instrumentIdentification: {
    id :`${scope}.instrumentIdentification`,
    defaultMessage : "Instrument Identification"
  },
  location: {
    id :`${scope}.location`,
    defaultMessage : "Location"
  },
  pollingInterval: {
    id :`${scope}.pollingInterval`,
    defaultMessage : "Polling service interval (sec)"
  },
  agentStatus: {
    id :`${scope}.agentStatus`,
    defaultMessage : "Agent Status"
  },
  modbusMapErrorMsg: {
    id :`${scope}.modbusMapErrorMsg`,
    defaultMessage : "Modbus Map is required"
  },
  incorrectfileErrorMsg: {
    id :`${scope}.incorrectfileErrorMsg`,
    defaultMessage : "File must have a comma (,) separated content in *.csv format."
  },
  devicetag: {
    id :`${scope}.devicetag`,
    defaultMessage : "Device Tag"
  },
  port: {
    id :`${scope}.port`,
    defaultMessage : "Port"
  },
  secretKey: {
    id :`${scope}.secretKey`,
    defaultMessage : "Secret Key"
  },
  dtsPollType: {
    id :`${scope}.dtsPollType`,
    defaultMessage : "Type of data being polled"
  },
});
