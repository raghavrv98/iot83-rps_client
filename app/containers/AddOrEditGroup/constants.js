/*
 *
 * AddOrEditGroup constants
 *
 */

export const ON_SUBMIT_REQUEST = "app/AddOrEditGroup/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddOrEditGroup/ON_SUBMIT_REQUEST_FAILURE";

export const GET_GROUP_DETAILS = "app/AddOrEditGroup/GET_GROUP_DETAILS";
export const GET_GROUP_DETAILS_SUCCESS = "app/AddOrEditGroup/GET_GROUP_DETAILS_SUCCESS";
export const GET_GROUP_DETAILS_FAILURE = "app/AddOrEditGroup/GET_GROUP_DETAILS_FAILURE";
