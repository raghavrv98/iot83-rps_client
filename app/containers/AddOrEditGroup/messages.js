import {defineMessages} from 'react-intl';

export const scope = "app.containers.AddOrEditGroup";

export default defineMessages({
    title:{
        id:`${scope}.title`,
        defaultMessage:"Manage Groups"
    },
    titleCreateGroup:{
        id:`${scope}.titleCreateGroup`,
        defaultMessage:"Create Group"
    },
    headingAdd:{
        id:`${scope}.headingAdd`,
        defaultMessage:"Create Group"
    },
    headingEdit:{
        id:`${scope}.headingEdit`,
        defaultMessage:" Edit Group"
    },
    formName:{
        id:`${scope}.formName`,
        defaultMessage:"Group Name"
    },
    formDescription:{
        id:`${scope}.formDescription`,
        defaultMessage:"Group Description"
    },
    groupAdded:{
        id:`${scope}.groupAdded`,
        defaultMessage:"Group added successfully."
    },
    groupUpdate:{
        id:`${scope}.groupUpdate`,
        defaultMessage:"Group Updated successfully."
    }
})