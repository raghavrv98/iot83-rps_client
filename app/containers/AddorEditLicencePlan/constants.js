/*
 *
 * AddorEditLicencePlan constants
 *
 */

export const DEFAULT_ACTION = "app/AddorEditLicencePlan/DEFAULT_ACTION";

export const ON_SUBMIT_REQUEST = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST";
export const ON_SUBMIT_REQUEST_SUCCESS = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_SUCCESS";
export const ON_SUBMIT_REQUEST_FAILURE = "app/AddorEditLicencePlan/ON_SUBMIT_REQUEST_FAILURE";

export const GET_PLANS_LIST = "app/AddorEditLicencePlan/GET_PLANS_LIST";
export const GET_PLANS_LIST_SUCCESS = "app/AddorEditLicencePlan/GET_PLANS_LIST_SUCCESS";
export const GET_PLANS_LIST_FAILURE = "app/AddorEditLicencePlan/GET_PLANS_LIST_FAILURE";