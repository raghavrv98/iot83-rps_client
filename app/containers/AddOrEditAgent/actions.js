/*
 *
 * AddOrEditAgent actions
 *
 */

import { CREATE_AGENT_REQUEST,GET_AGENT_DETAILS } from "./constants";

export function createAgent(data) {
  return {
    type: CREATE_AGENT_REQUEST,
    payload: data,
  };
}
export function getAgentDetails(id) {
  return {
    type: GET_AGENT_DETAILS,
    id
  };
}
