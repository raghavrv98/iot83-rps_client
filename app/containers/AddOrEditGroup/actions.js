/*
 *
 * AddOrEditGroup actions
 *
 */

import { ON_SUBMIT_REQUEST,GET_GROUP_DETAILS } from "./constants";

export function onSubmitHandler(payload,id) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload,
    id
  };
}

export function getGroupDetails(id) {
  return {
    type: GET_GROUP_DETAILS,
    id
  };
}
