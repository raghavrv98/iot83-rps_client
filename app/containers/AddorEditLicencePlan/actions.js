/*
 *
 * AddorEditLicencePlan actions
 *
 */

import { DEFAULT_ACTION, ON_SUBMIT_REQUEST, GET_PLANS_LIST } from "./constants";

export function defaultAction() {
  return {
    type: DEFAULT_ACTION
  };
}

export function onSubmitHandler(payload) {
  return {
    type: ON_SUBMIT_REQUEST,
    payload
  };
}

export function getPlansList(){
  return{
    type: GET_PLANS_LIST
  }
}