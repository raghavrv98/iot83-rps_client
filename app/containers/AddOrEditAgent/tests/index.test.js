import React from "react";
import { shallow } from "enzyme";
import history from "utils/history";
import { createAgent, getAgentDetails } from "../actions";
import { AddOrEditAgent, mapDispatchToProps } from "../index";
import { CREATE_AGENT_REQUEST, GET_AGENT_DETAILS } from "../constants";
import { FormattedMessage } from 'react-intl';
import messages from '../messages';
// import { fireEvent } from "@testing-library/react";

const match = {
  params: {
    id: ""
  }
};

const agentDataSuccess = {
  rps:{
    name: "",
    deviceId: "",
    type: "DCSAgent",
    agentKey: "",
    active: false,
  }
};


const agentDataFailure = "this is submit error";
const agentSuccess = "this is group details success";
const agentFailure = "this is group details Error";

describe("<AddOrEditAgent />", () => {

  it('match snapshots', () => {
    const wrap = shallow(<AddOrEditAgent match={match} createAgent={createAgent} getAgentDetails={getAgentDetails} />)
    expect(wrap).toMatchSnapshot()
  })
  
  it("clicking back button trigger onClick event", async () => {
    let expectedCount = history.length + 1;
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} createAgent={createAgent} />);
    let backbutton = wrapper.find("span.cursor-pointer");
    expect(backbutton.length).toBe(1);
    backbutton.simulate("click");
    expect(history.length).toEqual(expectedCount);
    expect(history.location.pathname).toEqual("/manageAgents");
  });


  it("clicking submit button trigger on having payload onClick event", async () => {
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} createAgent={createAgent} />);
    const onSubmitHandlerSpy = jest.spyOn(wrapper.instance(), "onSubmitHandler");
    wrapper.setState({ payload: {agentKey: "sample"}});
    wrapper.update();
    wrapper.instance().forceUpdate();
    let addOrEditForm = wrapper.find("form.contentForm");
    expect(addOrEditForm.length).toBe(1);
    addOrEditForm.simulate('submit', { preventDefault() { } });
    expect(onSubmitHandlerSpy).toHaveBeenCalledTimes(1);
  });

  it("clicking submit button trigger onClick event", async () => {
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} createAgent={createAgent} />);
    const onSubmitHandlerSpy = jest.spyOn(wrapper.instance(), "onSubmitHandler");
    wrapper.update();
    wrapper.instance().forceUpdate();
    let addOrEditForm = wrapper.find("form.contentForm");
    expect(addOrEditForm.length).toBe(1);
    addOrEditForm.simulate('submit', { preventDefault() { } });
    expect(onSubmitHandlerSpy).toHaveBeenCalledTimes(1);
  });

  it("test component will recieve props", async () => {
    spyOn(AddOrEditAgent.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} createAgent={createAgent} agentDataSuccess="" agentDataFailure="" agentSuccess="" agentFailure="" />);
    wrapper.setProps({ agentDataSuccess, agentDataFailure, agentSuccess, agentFailure });
    expect(AddOrEditAgent.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  });


  it('should test component when recieve props', () => {
    spyOn(AddOrEditAgent.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<AddOrEditAgent
      history={history}
      match={match}
      createAgent={createAgent}
      agentDataSuccess=""
      agentDataFailure=""
      agentSuccess=""
      agentFailure=""
    />);
    expect(wrapper).toBeDefined();
    wrapper.setState({ createAgent });
    wrapper.setProps({
      createAgent: { createAgent },
      agentDataSuccess:"",
      agentDataFailure:"",
      agentSuccess:"",
      agentFailure:""
    });
    expect(AddOrEditAgent.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  })



  it("test component will recieve props in edit case", async () => {
    const match = {
      params: {
        id: "demoId"
      }
    };
    spyOn(AddOrEditAgent.prototype, 'componentWillReceiveProps').and.callThrough();
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} getAgentDetails={getAgentDetails} createAgent={createAgent} agentDataSuccess="" agentDataFailure="" agentSuccess="" agentFailure="" />);
    wrapper.setProps({ agentDataSuccess, agentDataFailure, agentSuccess, agentFailure });
    expect(AddOrEditAgent.prototype.componentWillReceiveProps).toHaveBeenCalledTimes(1);
  });

  it("test edit case", async () => {
    const match = {
      params: {
        id: "demoId"
      }
    };
    const getAgentDetails = jest.fn();
    spyOn(AddOrEditAgent.prototype, 'componentDidMount').and.callThrough();  
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} getAgentDetails={getAgentDetails} />);
    expect(AddOrEditAgent.prototype.componentDidMount).toHaveBeenCalledTimes(1);
    expect(getAgentDetails).toHaveBeenCalledTimes(1);
  });

  test('Input should call handleChange on name change event', () => {
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} getAgentDetails={getAgentDetails} />);
    const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), "inputChangeHandler");
    wrapper.update();
    wrapper.instance().forceUpdate();
    wrapper.find('[id="name"]').simulate('change',{
      target: {id: "name", value: "name"}
    });
    expect(wrapper.find('[id="name"]').props().value).toEqual("name");
    expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
  })

  test('Input should call handleChange on type APSensingDTSPoller change event', () => {
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} getAgentDetails={getAgentDetails} />);
    const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), "inputChangeHandler");
    wrapper.update();
    wrapper.instance().forceUpdate();
    wrapper.find('[id="type"]').simulate('change',{
      target: {id: "type", value: "APSensingDTSPoller"}
    });
    expect(wrapper.find('[id="type"]').props().value).toEqual("APSensingDTSPoller");
    expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
  })

  test('Input should call handleChange on type DCSAgent change event', () => {
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} getAgentDetails={getAgentDetails} />);
    const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), "inputChangeHandler");
    wrapper.update();
    wrapper.instance().forceUpdate();
    wrapper.find('[id="type"]').simulate('change',{
      target: {id: "type", value: "DCSAgent"}
    });
    expect(wrapper.find('[id="type"]').props().value).toEqual("DCSAgent");
    expect(inputChangeHandlerSpy).toHaveBeenCalledTimes(1);
  })

  test('Input should call modalClose on click event', () => {
    let expectedCount = history.length + 1;
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} />);
    wrapper.setState({ modalFor: "submitSuccess" })
    wrapper.instance().modalCloseHandler();
    expect(history.length + 1).toEqual(expectedCount);
    expect(history.location.pathname).toEqual("/manageAgents");
  })

  test('Input should call modalClose on click event', () => {
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} />);
    wrapper.instance().modalCloseHandler();
    expect(wrapper.instance().state.isOpen).toEqual(false);
  })

  test('test inputChangeHandler select change ', () => {
    const wrapper = shallow(<AddOrEditAgent history={history} match={match} />);
    const inputChangeHandlerSpy = jest.spyOn(wrapper.instance(), 'inputChangeHandler');
    wrapper.setState({payload: {APSensingDTSPoller: {channel:[{}]}, type: "APSensingDTSPoller"}})
    let startPositionInput = wrapper.find('select#type').simulate('change',{
      target: {id: "activeChannels", value: "1"}
    }, 0);
    expect(startPositionInput.length).toBe(1);
  }) 

  // test('test nameChangeHandler input change ', () => {
  //   const wrapper = shallow(<AddOrEditAgent history={history} match={match} />);
  //   wrapper.setState({ payload: { APSensingDTSPoller: { channel: [{}] }, type: "APSensingDTSPoller" } })
  //   let startPositionInput = wrapper.find('input#ipAddress').simulate('change', {
  //     target: { id: "ipAddress", value: "1" }
  //   }, 0);
  //   expect(startPositionInput.length).toBe(1);
  // }) 

  it('Function testing on mapDispatchToProps', () => {
    const payload = "demo"
    const mockParams = {
      params: {
        id: "demoId"
      }
    }
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).createAgent(payload);
    expect(dispatch.mock.calls[0][0]).
      toEqual({
        type: CREATE_AGENT_REQUEST,
        payload: "demo"
      })
    mapDispatchToProps(dispatch).getAgentDetails(mockParams.params.id);
    expect(dispatch.mock.calls[1][0]).
      toEqual({
        type: GET_AGENT_DETAILS,
        id: "demoId"
      })
  });

  it('should render the page message', () => {
    const match = {
      params: {
        id: "demoId"
      }
    };
    const getAgentDetails = jest.fn();
    const renderedComponent = shallow(<AddOrEditAgent match={match} getAgentDetails={getAgentDetails} />);
    expect(
      renderedComponent.contains(<FormattedMessage {...messages.headingEdit} children={(message => message)} />),
    ).toEqual(true);
  });
});

function createDtWithFiles(files = []) {
  return {
    dataTransfer: {
      files,
      items: files.map(file => ({
        kind: 'file',
        size: file.size,
        type: file.type,
        getAsFile: () => file
      })),
      types: ['Files']
    }
  }
}

function createFile(name, size, type) {
  const file = new File([], name, { type })
  Object.defineProperty(file, 'size', {
    get() {
      return size
    }
  })
  return file
}