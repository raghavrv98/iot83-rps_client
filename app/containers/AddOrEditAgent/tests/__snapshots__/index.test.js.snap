// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddOrEditAgent /> match snapshots 1`] = `
<div
  className="appContent"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
       AddOrEditAgent 
    </title>
    <meta
      content="Description of AddOrEditAgent"
      name="description"
    />
  </HelmetWrapper>
  <div
    className="pageBreadcrumb"
  >
    <div
      className="flex-item fx-b70"
    >
      <p
        className="pd-l-30"
      >
        <span
          className="cursor-pointer"
          onClick={[Function]}
        >
          <button
            className="btn btn-transparent"
            type="button"
          >
            <i
              className="far fa-long-arrow-left"
            />
          </button>
          <FormattedMessage
            defaultMessage="Manage Agents"
            id="app.common.messages.manageAgents"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
        </span>
      </p>
      <h5>
        <FormattedMessage
          defaultMessage="Create Agent"
          id="app.containers.AddOrEditAgent.headingAdd"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </h5>
    </div>
    <div
      className="flex-item fx-b30 text-right align-items-center"
    />
  </div>
  <form
    className="contentForm"
    onSubmit={[Function]}
  >
    <h5
      className="formHeader"
    >
      <FormattedMessage
        defaultMessage="Create Agent"
        id="app.containers.AddOrEditAgent.headingAdd"
        values={Object {}}
      >
        <Component />
      </FormattedMessage>
      <div
        className="headerItemRight"
      >
        <div
          className="switchButtonLabel"
        >
          <label
            className="switchLabelText"
          >
            <FormattedMessage
              defaultMessage="Agent Status"
              id="app.containers.AddOrEditAgent.agentStatus"
              values={Object {}}
            >
              <Component />
            </FormattedMessage>
             :
          </label>
          <label
            className="switchLabel"
          >
            <input
              checked={false}
              id="active"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="switchMark"
            />
          </label>
        </div>
      </div>
    </h5>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="name"
          id="app.common.messages.name"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <input
        autoFocus={true}
        className="form-control"
        id="name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Device Tag"
          id="app.containers.AddOrEditAgent.devicetag"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <input
        className="form-control"
        id="deviceTag"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Type"
          id="app.common.messages.type"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <select
        className="form-control"
        disabled={false}
        id="type"
        onChange={[Function]}
        required={true}
        value="dcsagent"
      >
        <FormattedMessage
          defaultMessage="AP Sensing DTS Poller"
          id="app.containers.AddOrEditAgent.option1"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <FormattedMessage
          defaultMessage="DCS Agent"
          id="app.containers.AddOrEditAgent.option2"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </select>
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Modbus Port"
          id="app.containers.AddOrEditAgent.addModbusPort"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <input
        className="form-control"
        id="modbusPort"
        max="65535"
        min="1"
        onChange={[Function]}
        required={true}
        type="number"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        <FormattedMessage
          defaultMessage="Modbus Map"
          id="app.containers.AddOrEditAgent.addModbusMap"
          values={Object {}}
        />
        <sup
          className="mr-r-7"
        >
          <i
            className="fas fa-asterisk"
          />
        </sup>
        :
      </label>
      <div
        className="fx-b80"
      >
        <div
          className="dropZoneOuterBox pd-r-0 mr-0 w-100"
        >
          <n
            accept=".csv"
            className="dropZoneBox"
            disableClick={false}
            disablePreview={false}
            disabled={false}
            getDataTransferItems={[Function]}
            inputProps={Object {}}
            maxSize={Infinity}
            minSize={0}
            multiple={false}
            onDrop={[Function]}
            preventDropOnDocument={true}
          >
            <div
              className="dropZoneBoxContent"
            >
              <img
                src="IMAGE_MOCK"
              />
              <p>
                 
                <FormattedMessage
                  defaultMessage="Drop files or click to select files to upload."
                  id="app.common.messages.dropzonemsg"
                  values={Object {}}
                />
              </p>
            </div>
          </n>
        </div>
        <button
          className="btn-transparent text-cyan mr-t-8 pd-0"
          onClick={[Function]}
          type="button"
        >
          <FormattedMessage
            defaultMessage="Download Sample"
            id="app.common.messages.downloadSample"
            values={Object {}}
          >
            <Component />
          </FormattedMessage>
          <i
            className="far fa-download mr-l-8"
          />
        </button>
      </div>
    </div>
    <div
      className="form-group justify-content-end mt-3"
    >
      <button
        className="btn btn-danger"
        id="saveAgent"
      >
        <i
          className="far fa-check-circle"
        />
        <FormattedMessage
          defaultMessage="Save"
          id="app.common.messages.save"
          values={Object {}}
        >
          <Component />
        </FormattedMessage>
      </button>
    </div>
  </form>
</div>
`;
